File: 3213/3213.cs
public class Solution {
    public int ConstructStringWithMinimumCost(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(ConstructStringWithMinimumCost(root.left), ConstructStringWithMinimumCost(root.right));
    }
}

File: 3213/3213.go
func constructStringWithMinimumCost(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := constructStringWithMinimumCost(root.Left)
	r := constructStringWithMinimumCost(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}